### PR TITLE
Bugfix/default ns

### PR DESCRIPTION
**Pull request by bdmogal**
- [x] Ensures that the default namespace exists asynchronously during ``AppFabricServer`` startup.
- [x] Creates default namespace in tests that do not start ``AppFabricServer``

Jira: [CDAP-1213](https://issues.cask.co/browse/CDAP-1213)
Build: http://builds.cask.co/browse/CDAP-RBT144-2

### CODE REVIEW

File: cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/runtime/schedule/SchedulerTestBase.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+import co.cask.cdap.proto.ProgramType;
+import com.google.common.base.Predicate;
+
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Base class for scheduler tests
+ */
+public class SchedulerTestBase {
+  protected void waitUntilFinished(ProgramRuntimeService runtimeService,
```
Comment (anwar6953) [inline cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/runtime/schedule/SchedulerTestBase.java]: why not put common fields (such as Store, NamespaceAdmin) in here?
And also make it an abstract class?

Comment (bdmogal) [inline cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/runtime/schedule/SchedulerTestBase.java]: A lot more than that needs to be moved here - https://issues.cask.co/browse/CDAP-1807.

Wanted to get a review for these changes, then do it in a separate PR.

Comment (anwar6953) [inline cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/runtime/schedule/SchedulerTestBase.java]: Ah right; ok.

**Comment by bdmogal** [comment]
Build is green - http://builds.cask.co/browse/CDAP-RBT144-4

**Comment by anwar6953** [comment]
lgtm :+1:

**Comment by bdmogal** [comment]
Thanks @anwar6953. Merging.
